bpf/lib/nat.h

```diff
@@ -312,24 +313,28 @@ snat_v4_nat_handle_mapping(struct __ctx_buff *ctx,
 	if (*state) {
 		int ret;
 		struct ipv4_ct_tuple rtuple = {};
-
-		if (target->addr == (*state)->to_saddr &&
+		/* Check for the reverse SNAT entry. If it is missing (e.g. due to LRU
```

Comment:
Changes seem reasonable, I'm a little bit concerned with any possible regressions here resulting in unnecessary/unseen recreations of nat mappings.

Perhaps it would make sense to make sure there's some coverage of this particular scenario over in bpf/tests/bpf_nat_tests.c?

Reply:
@tommyp1ckles Thanks for the comment.  I was thinking what could be an unnoticed recreation? I am assuming fwd tuple and rev tuple should always co-exist. The only case is when they become orphaned, we don't need to recreate them and it is OK to stay broken. But this path won't touch the orphaned entries.

Reply:
Added the tests.